api/v1alpha1/release_types.go

```diff
@@ -156,20 +158,20 @@ func (r *Release) IsDone() bool {
 	return condition != nil && condition.Status != metav1.ConditionUnknown
 }
 
-// MarkDeployed sets the AllComponentsDeployed status in the Release to True or False with the provided reason and message.
-func (r *Release) MarkDeployed(status metav1.ConditionStatus, reason, message string) {
-	if status != metav1.ConditionUnknown {
+// MarkDeployed sets the AllComponentsDeployed status in the Release to True with the provided reason and message.
+func (r *Release) MarkDeployed(reason, message string) {
+	r.setStatusConditionWithMessage(applicationapiv1alpha1.ComponentDeploymentConditionAllComponentsDeployed,
+		metav1.ConditionTrue, ReleaseReason(reason), message)
+}
+
+// MarkDeploying sets the AllComponentsDeployed status in the Release to Unknown or False with the provided reason and message.
+func (r *Release) MarkDeploying(status metav1.ConditionStatus, reason, message string) {
+	if status != metav1.ConditionTrue {
```

Comment:
I don't think we should pass the status here. We should set it to False as that's the current behaviour from GitOps. If they fix it, this will be changed to Unknown.

Reply:
Why change the code to not support one of the condition types when as it is written, all three are supported? Who is going to watch gitops github repo to check for if/when they make the change?

Reply:
Because `False` should be a final state. I think this is something wrong in the GitOps side of the code. We could contribute a fix there.

Reply:
We can leave it as it is. If so, could you at least add a comment saying in some nice and kind words that the condition set by GitOps should provide an Unknown status and, as they don't, we have to consider False as a transient status? Or something properly written.

Reply:
Just for us to remember the reason and that we could/should provide a proper fix in the GitOps service.

Reply:
let me know (or just resolve) if what I wrote captures what you'd like written